src/support/bitset.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_support_bitset_h
+#define wasm_support_bitset_h
+
+#include <cassert>
+#include <cstddef>
+#include <iterator>
+#include <vector>
+
+namespace wasm {
+
+// Represent a set of integers in [0, N) with a bitvector of size N, where a 1
+// bit signifies set membership. This interface is intended to match that of
+// std::set or std::unordered_set as closely as possible.
+class BitSet : std::vector<bool> {
+  std::vector<bool>& vec() noexcept { return *this; }
+  const std::vector<bool>& vec() const noexcept { return *this; }
+
+  void ensure(size_t value) noexcept {
+    if (vec().size() <= value) {
+      vec().resize(value + 1);
+    }
+  }
+
+public:
+  using key_type = size_t;
+  using value_type = size_t;
+
+  class iterator {
+    const std::vector<bool>* parent = nullptr;
+    size_t index = 0;
+
+    iterator(const std::vector<bool>* parent) : parent(parent) {
+      const auto size = parent->size();
+      while (index < size && !(*parent)[index]) {
+        ++index;
+      }
```

Comment:
Looks like this code advances to the first item in the set? A comment might help.

Reply:
Yeah, I'll pull this out into a separate PR and add comments, as you suggest.